core/src/main/java/edu/wpi/grip/core/serialization/Project.java

```diff
@@ -78,10 +85,24 @@ public void open(File file) throws IOException {
         this.file = Optional.of(file);
     }
 
+    /**
+     * Loads the project defined by the given XML string. This is intended to be used to be able to programmatically
+     * run a pipeline from a remote source. Therefore, this does <strong>not</strong> save the contents to disk;
+     * if it is called in GUI mode, the user will have to manually save the file.
+     *
+     * @param projectXml the XML string defining the project to open
+     */
+    public void open(String projectXml) {
+        open(new StringReader(projectXml));
+    }
+
     @VisibleForTesting
     void open(Reader reader) {
+        eventBus.post(new ProjectUnloadedEvent());
         this.pipeline.clear();
         this.xstream.fromXML(reader);
+        eventBus.post(new ProjectLoadedEvent());
```

Comment:
This seems like a level of indirection. Could you just call a method on an object instead??
What is the goal of these events? Are these events only used in one place?